test/end2end_test.go

```diff
@@ -6930,14 +6947,14 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 		}
 		writer.Flush() // necessary since client is expecting preface before declaring connection fully setup.
 		var sid uint32
-		// Loop until conn is closed and framer returns io.EOF
+		// Loop until conn is closed and framer returns possible Stop error messages
 		for requestNum := 0; ; requestNum = (requestNum + 1) % len(s.responses) {
 			// Read frames until a header is received.
 			for {
 				frame, err := framer.ReadFrame()
 				if err != nil {
```

Comment:
Let's move the `nil` check into `errMsgContainsPossibleStopMessages` and simplify both these call sites  (remove this `if` and remove `err == nil ||` above.

Reply:
Yeah I think I misread this (the `return` outside the following `if` is important), so I think this part is fine now.  I left another comment at the other call site.